modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/util/LayoutCopyHelperImpl.java

```diff
@@ -177,27 +175,24 @@ public Layout copyLayout(
 			}
 		};
 
-		Callable<Layout> callable = new CopyLayoutCallable(
-			consumer, sourceLayout, segmentsExperiencesIds, targetLayout,
-			segmentsExperiencesIds);
-
-		boolean copyLayout = CopyLayoutThreadLocal.isCopyLayout();
```

Comment:
@JanisZhang Please don't inline these thread local related variables.

Basically the logic is,

```
// save the current values to variables

var a = xxx.getValue();

try {

    // set the new value and execute operation

    xxx.setValue(b);

    doSomething();
}
finally {

    // set the original value back

    xxx.setValue(a);
}
```

If you inline them, the logic will be broken.